src/MountedTraversal.js

```diff
@@ -177,6 +178,12 @@ export function parentsOfInst(inst, root) {
   return pathToNode(inst, root).reverse();
 }
 
+export function instMatchesObjectProps(inst, props) {
+  if (!isDOMComponent(inst)) return false;
+  const node = getNode(inst);
+  return !isEmpty(props) && isMatch(propsOfNode(node), props);
```

Comment:
I'll update this after we discuss the best solution to the nested objects below.